nemo_aligner/models/nlp/gpt/megatron_gpt_ppo_actor.py

```diff
@@ -280,19 +262,36 @@ def infer(self, inference_batch):
         prompt_lengths = inference_batch["length"].cuda(non_blocking=True)
         inputs = (prompt_tokens, prompt_lengths)
 
+        strategy = TrackLengthGPTModelTextGenerationStrategy(
+            model=self, context_lengths=prompt_lengths, max_length=self._length_params["max_length"]
+        )
         actor_output = self.generate(
-            inputs=inputs, length_params=self._length_params, sampling_params=self._sampling_params
+            inputs=inputs, length_params=self._length_params, sampling_params=self._sampling_params, strategy=strategy
         )
 
+        response_lengths = None
+        if parallel_state.is_pipeline_last_stage():
```

Comment:
(just a thought): the contract we have with nemo generate is that the outputs it returns will be duplicated across all ranks. 

This breaks in contract -- I think we should move this broadcast PP logic into the strategy itself(since the strategy is already checking if PP is last stage when computing get_length). 

then in the model we can just perform the `assert (response_lengths <= self.cfg.encoder_seq_length).all()` sanity check

Reply:
Good idea, I agree it's nice that way. Done in 02f0a612.